src/core_plugins/tile_map/public/base_maps_visualization.js

```diff
@@ -0,0 +1,196 @@
+import _ from 'lodash';
+import { KibanaMap } from './kibana_map';
+import 'ui/vis/map/service_settings';
+
+
+export function BaseMapsVisualizationProvider(serviceSettings) {
+
+  return class BaseMapsVisualization {
+
+    constructor(element, vis) {
+      this.vis = vis;
+      this._container = element;
+      this._kibanaMap = null;
+      this._baseLayerDirty = true;
+    }
+
+    destroy() {
+      if (this._kibanaMap) {
+        this._kibanaMap.destroy();
+      }
+    }
+
+    isDataNotUsable() {
+      return false;
+    }
+
+    async render(esResponse, status) {
+
+      return new Promise(async (resolve) => {
+
+        if (!this._kibanaMap) {
+          await this._makeKibanaMap();
+        }
+
+        if (status.resize) {
+          this._kibanaMap.resize();
+        }
+        if (status.params || status.aggs) {
+          await this._updateParams();
+        }
+
+        if (this.isDataNotUsable(esResponse)) {
+          return resolve();
+        }
+
+        if (status.data) {
+          await this._updateData(esResponse);
+        }
+        if (status.uiState) {
+          this._kibanaMap.useUiStateFromVisualization(this.vis);
+        }
+
+        this._doRenderComplete(resolve);
+
+      });
+    }
+
+    async _makeKibanaMap() {
+
+      if (this._kibanaMap) {
+        this._kibanaMap.destroy();
+      }
+      const options = _.clone({});
```

Comment:
We shouldn't need the `_.clone` here, and should just be able to do `const options = {}`.